app/javascript/packs/flagUserModal.js

```diff
@@ -0,0 +1,141 @@
+          <label for="vomit-all" class="crayons-field__label">
+            Make all posts by this author less visible
+            <p class="crayons-field__description">
+              This author consistently posts content that violates DEV's code of conduct because it is harassing, offensive or spammy.
+            </p>
+          </label>
+        </div>
+        <a href="/report-abuse?url=${document.location}" class="fs-base abuse-report-link">Report other inappropriate conduct</a>
+      </div>
+      <div class="buttons-container">
+        <a href="#" class="crayons-btn" id="confirm-flag-user-action">Confirm action</a>
+        <a href="#" class="crayons-btn" id="cancel-flag-user-action">Cancel</a>
+      </div>
+    </div>
+  </div>
+  <div class="crayons-modal__overlay"></div>
+</div>
+`;
+
+  if (user.id !== articleAuthorId && user.trusted) {
+    const modContainer = document.getElementById('mod-container');
+
+    document.querySelector(
+      '.modal-snackbar-container',
+    ).innerHTML = modalSnackbarHTML;
+    document.querySelector(
+      '.flag-user-modal-container',
+    ).innerHTML = flagUserModalHTML;
+    modContainer.addEventListener('load', () => {
+      modContainer.contentWindow.document
+        .getElementById('open-flag-user-modal')
+        .addEventListener('click', toggleFlagUserModal);
+    });
+  }
+
+  // Event listeners to Close the Modal
+  document
+    .querySelector('.crayons-modal__overlay')
+    .addEventListener('click', toggleFlagUserModal);
+
+  document
+    .querySelector('.modal-header-close-icon')
+    .addEventListener('click', toggleFlagUserModal);
+
+  document
+    .getElementById('confirm-flag-user-action')
+    .addEventListener('click', () => {
+      if (!document.getElementById('vomit-all').checked) {
+        alert('No Selection Made!');
+      } else {
+        toggleFlagUserModal();
+      }
+    });
+
+  document
+    .getElementById('cancel-flag-user-action')
+    .addEventListener('click', toggleFlagUserModal);
+
+  document
+    .getElementById('confirm-flag-user-action')
+    .addEventListener('click', (e) => {
+      e.preventDefault();
+      const vomitAllOption = document.getElementById('vomit-all');
+      const vomitAllSnackbar = document.getElementById('vomit-all-snackbar');
+
+      const flagUserApiCall = (body, snackbar) => {
+        getCsrfToken()
+          .then(sendFetch('reaction-creation', body))
+          .then((response) => {
+            if (response.status === 200) {
+              response.json().then(flashSnackbar(snackbar));
+            }
+          });
+      };
+
+      if (vomitAllOption.checked) {
+        const formData = new FormData();
+        formData.append('reactable_type', vomitAllOption.dataset.reactableType);
+        formData.append('category', vomitAllOption.dataset.category);
+        formData.append('reactable_id', vomitAllOption.dataset.reactableId);
+
+        flagUserApiCall(formData, vomitAllSnackbar);
```

Comment:
Hmm let's move this to JSON instead of using `FormData`, like so:
```
const body = JSON.stringify({
  reactable_type: vomitAllOption.dataset.reactableType,
  category: vomitAllOption.dataset.category,
  reactable_id: vomitAllOption.dataset.reactableId,
})
```

Reply:
Hi @Zhao-Andy, yeah I had tried this beforehand, but I kept getting this error:

```
Uncaught (in promise) TypeError: body.append is not a function
    at base.js:4212
```

When I investigated, I found the error pops up on the arrow'd line below:
```
const fetchCallback = ({ url, headers = {}, addTokenToBody = false, body }) => {
  return csrfToken => {
    if (addTokenToBody) {
      body.append('authenticity_token', csrfToken); <=====
    }
    return window.fetch(url, {
      method: 'POST',
      headers: {
        'X-CSRF-Token': csrfToken,
        ...headers,
      },
      body,
      credentials: 'same-origin',
    });
  };
};
```

Reply:
So I defaulted back to the way I saw it implemented in other use-cases that call that `fetchCallback` function.

Reply:
Ah I see. I think refactoring this to use the `request` function that Nick made would be good then. Here's an example from `actionsPanel.js`: https://github.com/thepracticaldev/dev.to/blob/1ebad88db61f7a54515f867e7546628d935ee45a/app/javascript/actionsPanel/actionsPanel.js#L35-L57

You'll need to add `import { request } from '../utilities/http/request';` at the top of this file.

Reply:
You mean refactoring the `fetchCallback` function? Or my function in `flagUserModal.js`?